eval/chat_benchmarks/arena_hard_auto/eval_instruct.py

```diff
@@ -0,0 +1,177 @@
+                                },
+                            ),
+                            idx,
+                        )
+                    )
+                except Exception as e:
+                    self.logger.error(f"Error preparing instance {idx}: {str(e)}")
+                    continue
+
+            with torch.no_grad():
+                self.logger.info("Generating responses for Arena Hard ...")
+                outputs = self.compute(model, all_instances)
+
+            if model.rank != 0:
+                return None
+
+            model_outputs = []
+            for idx, (example, output) in enumerate(zip(examples, outputs)):
+                try:
+                    encoding = tiktoken.encoding_for_model("gpt-3.5-turbo")
+                    instance = {
+                        "question_id": example["question_id"],
+                        "answer_id": shortuuid.uuid(),
+                        "model_id": model.model_identifier,
+                        ## TODO: fix token_len computation -- currently 1.3 * len(output)
+                        "choices": [{"index":0,"turns":[{"content": output,"token_len": len(encoding.encode(output, disallowed_special=()))}]}],
+                    }
+                    with open(f"eval/chat_benchmarks/arena_hard_auto/data/arena-hard-v0.1/model_answer/{model.model_identifier}.jsonl", "a") as f:
+                        f.write(json.dumps(instance, ensure_ascii=False) + "\n")
+                    model_outputs.append(instance)
+                except Exception as e:
+                    self.logger.error(f"Error processing output {idx}: {str(e)}")
+                    continue
+
+            self.logger.info(f"Generated {len(model_outputs)} responses")
+
+            return {"model_outputs": model_outputs, "model_identifier": model.model_identifier}
+
+        except Exception as e:
+            self.logger.error(f"Error in generate_responses: {str(e)}")
+            raise
+
+    def evaluate_responses(self, model_results: Dict) -> Dict[str, float]:
+        """
+        Evaluate the generated responses using Arena Hard evaluation metrics.
+
+        Args:
+            results: Dictionary containing model outputs and identifier
+
+        Returns:
+            Dictionary containing evaluation metrics
+        """
+        
+        self.logger.info("Running Arena Hard judgements...")
+        model_name = model_results['model_identifier']
+        print(f'model name: {model_name}')
+        print(f'annotator model: {self.annotator_model}')
+        execute_judgment(model_name = model_name, judge_model = self.annotator_model)
+        
+        ## save a leaderboard in leaderboard dir
+        subprocess.run(['python', '-m', 'eval.chat_benchmarks.arena_hard_auto.show_result', '--judge-name', f'{self.annotator_model}', '--output'])
+        df = pd.read_csv(f'eval/chat_benchmarks/arena_hard_auto/leaderboard/arena_hard_leaderboard_{self.annotator_model}.csv')
```

Comment:
Thanks a lot @asuvarna31 !
Can we just avoid read/write to file? seems like we can just use the input model_results here?

Reply:
Addressed your comments @neginraoof